modules/dxp/apps/search-experiences/search-experiences-web/src/main/resources/META-INF/resources/sxp_blueprint_admin/js/edit_sxp_blueprint/configuration_tab/index.js

```diff
@@ -152,47 +176,92 @@ function ConfigurationTab({
+
+											<ClayTooltipProvider>
+												<span
+													title={
+														!frameworkConfig.collectionProvider
+															? Liferay.Language.get(
+																	'enable-as-a-collection-provider-help'
+																)
+															: Liferay.Language.get(
+																	'disable-as-a-collection-provider-help'
+																)
+													}
+												>
+													<ClayIcon
+														className="c-ml-2 text-secondary"
+														symbol="question-circle-full"
+													/>
+												</span>
+											</ClayTooltipProvider>
+										</>
+									}
+									onToggle={_handleCollectionProviderChange}
+									toggled={
+										frameworkConfig.collectionProvider ||
+										false
+									}
+								/>
 
-										<ClayTooltipProvider>
-											<span
-												title={
-													!frameworkConfig.collectionProvider
-														? Liferay.Language.get(
-																'enable-as-a-collection-provider-help'
-															)
-														: Liferay.Language.get(
-																'disable-as-a-collection-provider-help'
-															)
-												}
-											>
-												<ClayIcon
-													className="c-ml-2 text-secondary"
-													symbol="question-circle-full"
-												/>
-											</span>
-										</ClayTooltipProvider>
-									</>
-								}
-								onToggle={_handleCollectionProviderChange}
-								toggled={
-									frameworkConfig.collectionProvider || false
-								}
-							/>
-
-							<span className="c-ml-2 sheet-text">
-								<LearnMessage resourceKey="collections-with-search-blueprints" />
-							</span>
-						</div>
+								<span className="c-ml-2 sheet-text">
+									<LearnMessage resourceKey="collections-with-search-blueprints" />
+								</span>
+							</div>
+
+							{isDefined(
+								frameworkConfig.legacyAssetCollectionProvider
+							) && (
+								<div className="align-items-center c-mb-4">
+									<ClayToggle
+										aria-label={Liferay.Language.get(
+											'enable-as-a-collection-provider-with-return-type-asset'
+										)}
+										disabled={
+											!frameworkConfig.collectionProvider
+										}
+										label={
+											<>
+												{Liferay.Language.get(
+													'enable-as-a-collection-provider-with-return-type-asset'
+												)}
+
+												<ClayTooltipProvider>
+													<span
+														title={Liferay.Language.get(
+															'enable-as-a-collection-provider-with-return-type-asset-help'
+														)}
+													>
+														<ClayIcon
+															className="c-ml-2 text-secondary"
+															symbol="question-circle-full"
+														/>
+													</span>
+												</ClayTooltipProvider>
+											</>
+										}
+										onToggle={
+											_handleLegacyAssetCollectionProviderChange
+										}
+										toggled={
+											frameworkConfig.legacyAssetCollectionProvider ||
+											false
+										}
```

Comment:
Just wanted to make sure you were aware of this: This is fine as-is since the fallback is `false`, but assuming you wanted to set the value of `frameworkConfig.legacyAssetCollectionProvider` if it's defined, I'd use `??` here. Because technically if `frameworkConfig.legacyAssetCollectionProvider` is `false`, this fallbacks to the value after the `||`.

Reply:
Oh right! Yea I'll add `??` instead